pkg/controller/machinepool/actuator.go

```diff
@@ -19,4 +21,10 @@ type Actuator interface {
 	// or not, and an error. The boolean may be set in situations where we have not encountered an error, but still need
 	// to wait before we can proceed with reconciling. (e.g. obtaining a pool name lease)
 	GenerateMachineSets(*hivev1.ClusterDeployment, *hivev1.MachinePool, log.FieldLogger) (msets []*machineapi.MachineSet, proceed bool, genError error)
+
+	// GenerateCAPIMachineSets generates upstream CAPI MachineSets
+	GenerateCAPIMachineSets(*hivev1.ClusterDeployment, *hivev1.MachinePool, log.FieldLogger) (msets []*capiv1.MachineSet, proceed bool, genError error)
+
+	// GenerateMachineTemplates()
+	GenerateMachineTemplates(*hivev1.ClusterDeployment, *hivev1.MachinePool, log.FieldLogger) (templates interface{}, proceed bool, genError error)
```

Comment:
what do you think about 2 separate interfaces, one for generating remote machinesets, another for local clusterapi machinesets ?

Reply:
I like the idea of separating them. This was hard to fit into one.

Reply:
also i'm thinking what is the use of GenerateMachineTemplates function in the general interface.

machine templates seems like an additional object you need for a capi-machineset. so should we update the GenerateCAPIMachineSets to return a list of opaque k8s objects... that need to be created for the machineset to complete. 
i'm thinking there isn't specific handling we intend to do with machine templates in actuators other than creating them, so seems like not useful for actuator implementation expose something for that.